pkg/configmap/configmap_writer.go

```diff
@@ -0,0 +1,222 @@
+package configmap
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/ghodss/yaml"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/sirupsen/logrus"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/operator-framework/operator-registry/pkg/client"
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+)
+
+// configmap keys can contain underscores, but configmap names can not
+var unallowedKeyChars = regexp.MustCompile("[^-A-Za-z0-9_.]")
+
+const (
+	EnvContainerImage           = "CONTAINER_IMAGE"
+	ConfigMapImageAnnotationKey = "olm.sourceImage"
+)
+
+type AnnotationsFile struct {
+	Annotations struct {
+		Resources      string `json:"operators.operatorframework.io.bundle.manifests.v1"`
+		MediaType      string `json:"operators.operatorframework.io.bundle.mediatype.v1"`
+		Metadata       string `json:"operators.operatorframework.io.bundle.metadata.v1"`
+		Package        string `json:"operators.operatorframework.io.bundle.package.v1"`
+		Channels       string `json:"operators.operatorframework.io.bundle.channels.v1"`
+		ChannelDefault string `json:"operators.operatorframework.io.bundle.channel.default.v1"`
+	} `json:"annotations"`
+}
+
+type ConfigMapWriter struct {
+	manifestsDir  string
+	configMapName string
+	namespace     string
+	clientset     *kubernetes.Clientset
+}
+
+func NewConfigMapLoaderForDirectory(configMapName, namespace, manifestsDir, kubeconfig string) *ConfigMapWriter {
```

Comment:
nit: `NewConfigMapLoaderForDirectory`

nit: might want to use the functional option pattern for arguments we've been using elsewhere